src/main/java/htsjdk/samtools/reference/HadoopReferenceSequenceFileFactory.java

```diff
@@ -20,15 +20,20 @@
 
     public static ReferenceSequenceFile getReferenceSequenceFile(Path fastaFile) throws IOException {
         Path fastaIndexFile = getFastaIndexFileName(fastaFile);
+        Path fastaDictFile = getFastaDictFileName(fastaFile);
 
         File tempDir = Files.createTempDir();
         File localFakeFastaFile = new File(tempDir, fastaFile.getName());
         File localFastaIndexFile = new File(tempDir, fastaIndexFile.getName());
+        File localFastaDictFile = new File(tempDir, fastaDictFile.getName());
 
         localFakeFastaFile.createNewFile(); // create empty file to satisfy IndexedFastaSequenceFile; we'll actually read from HDFS
 
         FileSystem fs = fastaFile.getFileSystem(new Configuration());
         fs.copyToLocalFile(fastaIndexFile, new Path(localFastaIndexFile.toString()));
+        if (fs.exists(fastaDictFile)) {
+            fs.copyToLocalFile(fastaDictFile, new Path(localFastaDictFile.toString()));
+        }
```

Comment:
Throw a `UserException` if there's no `.dict` file? (we traditionally have required one to be present for all fasta files)